Ops/src/OpsWorkerGenetic.cpp

```diff
@@ -147,66 +143,91 @@ void WorkerGenetic::setMutationProbability(double value)
     mMutationProbability = value;
 }
 
+//! @brief Set elites paremter
+void WorkerGenetic::setNumberOfElites(size_t value)
+{
+    mNumElites = value;
+}
+
 
 //! @brief Auxiliary function for selecting parents for crossover
 void WorkerGenetic::selectParents() {
-    double bestValue = std::numeric_limits<double>::max();
-    for(size_t i=0; i<mNumPoints; ++i) {
-        double value = mObjectives[i]*opsRand();
-        if(value < bestValue) {
-            mParent1 = i;
-            bestValue = value;
+    std::vector<size_t> sortedIds = getIdsSortedFromWorstToBest();
+
+    // Select parent 1
+    std::vector<size_t> competitors;
+    competitors.push_back(opsRand()*(mNumPoints));
+    competitors.push_back(opsRand()*(mNumPoints));
+    while(competitors.size() < 2) {
+        competitors.push_back(opsRand()*(mNumPoints));
+    }
+    size_t bestCompetitor = 0;
+    for(auto competitor : competitors) {
```

Comment:
For readability, break this out into a help function `findMaxOf(stuff)`   or similar.
It is also used below, or maybe there is a std::algorithm available

Reply:
Replaced with the `max_element` function.